### DISCUSSION TITLE
Comparing polytype with other methods

### DISCUSSION THREAD

**Discussion by AbdealiLoKo** [fasttime/Polytype#16] [Ideas]
I've been trying to figure out inheritance / mixin approach for the past few days.
And I've seen a few methods around

polytype seems to be the most robust approach I have come acorss based on the documentation so far.
I was wondering if there was some nice comparison of some of the common approaches and pitfalls or other approaches that could exist ?

For example -
1. The single inheritance available in JS now prototype replacement (which does not allow multiple inheritance)
2. The approach of modifying the prototype of classes - like `Object.assign(MyClass.prototype, MyMixin);` (which does not allow me to use super(), instanceof, doesnt handle typing, etc.)
3. The approach mentioned in [typescript's mixin documentation](https://www.typescriptlang.org/docs/handbook/mixins.html) right now about dynamic classes inherited in a function (Which has a bit of an awkward syntax ? But seems to handle everything else ?)
4. ts-mixer

And so on.

**Comment by fasttime** [comment]
I don't know any good resources for what you are asking, but if a well-written article existed I'd be happy to link to it in the documentation. Maybe under a new section *Alternatives*, or similar. Mixins are often a good solution - especially in TypeScript, where types are immutable - but there are more alternatives. I've converted the original issue into a discussion for now, let's see if we can collect more ideas. Thanks for opening this.
